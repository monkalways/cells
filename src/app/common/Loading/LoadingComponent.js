import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  size: PropTypes.number,
};

const defaultProps = {
  size: 100,
};

const LoadingComponent = ({ classes, size }) => (
  <div className={classes.container}>
    <CircularProgress className={classes.progress} size={size} />
  </div>
);

LoadingComponent.propTypes = propTypes;
LoadingComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  container: {
    display: 'flex',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  progress: {
    margin: theme.spacing.unit * 2,
  },
}))(LoadingComponent);
