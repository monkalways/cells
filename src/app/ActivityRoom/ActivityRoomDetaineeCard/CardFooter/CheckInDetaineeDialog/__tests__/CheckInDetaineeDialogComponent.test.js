import React from 'react';
import { shallow } from 'enzyme';
import { Dialog } from '@material-ui/core';

import CheckInDetaineeDialogComponent from '../CheckInDetaineeDialogComponent';
import Loading from '../../../../../common/Loading';

describe('CheckInDetaineeDialogComponent', () => {
  let isCheckingIn;
  let onCheckIn;
  let onClose;

  beforeEach(() => {
    isCheckingIn = true;
    onCheckIn = jest.fn();
    onClose = jest.fn();
  });

  const setup = () => {
    const detainee = {
      firstName: 'John',
      lastName: 'Smith',
      activityRoomName: 'T1',
    };
    const isDialogOpen = false;
    const usage = 'phone';

    return shallow(<CheckInDetaineeDialogComponent
      detainee={detainee}
      isDialogOpen={isDialogOpen}
      isCheckingIn={isCheckingIn}
      onCheckIn={onCheckIn}
      onClose={onClose}
      usage={usage}
    />);
  };

  it('should render', () => {
    isCheckingIn = false;
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Loading component when isCheckingIn is true', () => {
    const wrapper = setup();

    expect(wrapper.find(Loading)).toExist();
  });

  it('should call onCheckIn when checkInButton is clicked', () => {
    const wrapper = setup();

    wrapper.find('#checkInButton').simulate('click');
    expect(onCheckIn).toBeCalled();
  });

  it('should call onClose when cancelButton is clicked', () => {
    const wrapper = setup();

    wrapper.find('#cancelButton').simulate('click');
    expect(onClose).toBeCalled();
  });

  it('should call onClose when dialog is closed', () => {
    const wrapper = setup();

    wrapper.find(Dialog).simulate('close');
    expect(onClose).toBeCalled();
  });
});
