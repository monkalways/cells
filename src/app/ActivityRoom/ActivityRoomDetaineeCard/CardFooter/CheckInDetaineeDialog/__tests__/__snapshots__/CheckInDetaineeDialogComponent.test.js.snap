// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckInDetaineeDialogComponent should render 1`] = `
<WithStyles(Dialog)
  disableBackdropClick={false}
  disableEscapeKeyDown={false}
  onClose={[MockFunction]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Placing Smith, John in phone room T1?
  </WithStyles(DialogTitle)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      disabled={false}
      id="cancelButton"
      onClick={[MockFunction]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      disabled={false}
      id="checkInButton"
      onClick={[MockFunction]}
      variant="contained"
    >
      Confirm
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
